tests/registries/test_handler_matching.py

```diff
@@ -50,7 +52,118 @@ def cause_with_diff(resource):
+    pytest.param({'someannotation': 'somevalue', 'otherannotation': 'othervalue'}, id='with-extra-annotation'),
+])
+def cause_with_annotation_value(resource, request):
+    body = {'metadata': {'annotations': request.param}}
+    return Mock(resource=resource, event='some-event', diff=None, body=body)
+
+
+@pytest.fixture(params=[
+    pytest.param({}, id='without-annotation'),
+    pytest.param({'someannotation': 'othervalue'}, id='with-other-value'),
+    pytest.param({'otherannotation': 'othervalue'}, id='with-other-annotation'),
+])
+def cause_without_annotation_value(resource, request):
+    body = {'metadata': {'annotations': request.param}}
+    return Mock(resource=resource, event='some-event', diff=None, body=body)
+
+
+@pytest.fixture(params=[
+    pytest.param({'someannotation': 'somevalue'}, id='with-annotation'),
+    pytest.param({'someannotation': 'othervalue'}, id='with-other-value'),
+])
+def cause_with_annotation_exists(resource, request):
+    body = {'metadata': {'annotations': request.param}}
+    return Mock(resource=resource, event='some-event', diff=None, body=body)
+
+
+@pytest.fixture(params=[
+    pytest.param({}, id='without-annotation'),
+    pytest.param({'otherannotation': 'othervalue'}, id='with-other-annotation'),
+])
+def cause_without_annotation_exists(resource, request):
+    body = {'metadata': {'annotations': request.param}}
+    return Mock(resource=resource, event='some-event', diff=None, body=body)
+
+
+@pytest.fixture(params=[
+    pytest.param({}, id='without-annotation'),
+    pytest.param({'someannotation': 'somevalue'}, id='with-annotation'),
+    pytest.param({'someannotation': 'othervalue'}, id='with-other-value'),
+    pytest.param({'otherannotation': 'othervalue'}, id='with-other-annotation'),
+    pytest.param({'someannotation': 'somevalue', 'otherannotation': 'othervalue'}, id='with-extra-annotation'),
+])
+def cause_any_annotations(resource, request):
+    body = {'metadata': {'annotations': request.param}}
+    return Mock(resource=resource, event='some-event', diff=None, body=body)
+
+
+@pytest.fixture(params=[
+    pytest.param({'labels': {'somelabel': 'somevalue'}, 'annotations': {'someannotation': 'somevalue'}}, id='with-label-annotation'),
+    pytest.param({'labels': {'somelabel': 'somevalue', 'otherlabel': 'othervalue'}, 'annotations': {'someannotation': 'somevalue'}}, id='with-extra-label-annotation'),
+    pytest.param({'labels': {'somelabel': 'somevalue'}, 'annotations': {'someannotation': 'somevalue', 'otherannotation': 'othervalue'}}, id='with-label-extra-annotation'),
+    pytest.param({'labels': {'somelabel': 'somevalue', 'otherlabel': 'othervalue'}, 'annotations': {'someannotation': 'somevalue', 'otherannotation': 'othervalue'}}, id='with-extra-label-extra-annotation'),
+])
+def cause_labels_and_annotations(resource, request):
+    body = {'metadata': request.param}
+    return Mock(resource=resource, event='some-event', diff=None, body=body)
+
```

Comment:
If those fixtures are not reused across multiple tests, it might be more convenient and easier to put them (i.e. the simulated Cause mocks) directly into the test functions below, and parametrize those tests with `@pytest.mark.parametrize(...)`.

That would also make the tests easier to read, as initialisation, action, and assertion will be gathered in one place within ~5 lines span.